### DISCUSSION TITLE
Structure/hierarchy changed - single language

### DISCUSSION THREAD

**Discussion by jehoshua7** [razonyang/hugo-theme-bootstrap#440] [Q&A]
I have a small bash script to `rm` everything locally, then `git submodule add https://github.com/razonyang/hugo-theme-bootstrap themes/hugo-theme-bootstrap`, and copy all the content/posts from a backup. It was working fine up until the end of Sept. 2021. Today it had problems. So the version I last used was 28th Sept. 2021, and now as I browse the theme I can see the file structure has changed , but only minor.

I don't use multi-languages, and the (old) structure was /hugo-theme-bootstrap/exampleSite/content/en/

Are there any changes I may have to do locally now ?  I only have posts, but do use:

/archives
/categories
/series
/tags

in the deployment. Are there any configuration changes since Sept. 21, as I used to have config files backed up and simply copied them across on a reload.

**Comment by razonyang** [comment]
The main breaking changes is that we use Hugo Pipes #405 for generating assets. See also [CHANGELOG](https://github.com/razonyang/hugo-theme-bootstrap/blob/master/CHANGELOG.md) and [UPGRADE](https://github.com/razonyang/hugo-theme-bootstrap/blob/master/UPGRADE.md).

If you're still failing to build your site, please provide the details.

**Reply by jehoshua7** [reply]
@razonyang thanks. I did look at both those documents prior to posting this discussion question. I assumed as I was completely removing the theme locally, then adding from 'git' that all those changes will already be reflected. I just checked a few and they seem to be already reflected in the latest version of this theme.

I also rolled back to Hugo version v0.88.1-5BC54738 , as quite a few messages about 

> js.Build failed: Could not resolve 

when I tried `hugo server`, however even with hugo v0.94.0-63B23660 , the same messages when attempting the server parameter. The structure changes I can cater for, but the JS build means no testing yet for the latest Hugo and the latest bootstrap theme.

**Reply by razonyang** [reply]
I just commented https://github.com/razonyang/hugo-theme-bootstrap/discussions/440#discussioncomment-2336176. Please feel free to comment if you're still failing to build the site.

**Comment by razonyang** [comment]
I just realized that you used `exampleSite/content/en/` before. You can disable other languages by setting the following configuration(`config.toml`). In this way, you can easily set up a single language demo site without deleting any files.

```toml
defaultContentLanguageInSubdir = false
disableLanguages = ["fr", "zh-cn" ,"zh-tw"]
```

Since we use Hugo Pipes, you'll need to install **extended** Hugo and **npm** for building your site:

```
$ hugo mod npm pack
$ npm install
$ hugo
```

**Reply by jehoshua7** [reply]
In my `config.toml` I have

> # Multilingual mode
defaultContentLanguage = "en"
#defaultContentLanguageInSubdir = true # If you use only one language comment this option
disableLanguages = ["fr", "zh-cn" ,"zh-tw"] # For disable one or more language

I have done the extended Hugo and npm as per your instructions. There were many warnings in the `npm install` such as

> npm WARN notsup Unsupported engine for d3-brush@3.0.0: wanted: {"node":">=12"} (current: {"node":"10.19.0","npm":"6.14.4"})
npm WARN notsup Not compatible with your version of node/npm: d3-brush@3.0.0

When I run `hugo` it looks like this

![image](https://user-images.githubusercontent.com/33979036/157798780-019778e1-4a10-4ec6-af37-f35804902746.png)

**Reply by razonyang** [reply]
Please put the following configuration into `config.toml`, and then try again.

```toml
[build]
  writeStats = true
```

**Comment by jehoshua7** [comment]
I'll check my (backup) config files against the themes config files and see if there is anythng obvious

**Comment by jehoshua7** [comment]
I followed the instructions at https://hbs.razonyang.com/en/docs/installation/git-submodule/#installation , and put it all in a new path called "myblog". This part had warnings

`npm install`

> npm WARN notsup Unsupported engine for copy-webpack-plugin@10.2.4: wanted: {"node":">= 12.20.0"} (current: {"node":"10.19.0","npm":"6.14.4"})
npm WARN notsup Not compatible with your version of node/npm: copy-webpack-plugin@10.2.4
npm WARN notsup Unsupported engine for schema-utils@4.0.0: wanted: {"node":">= 12.13.0"} (current: {"node":"10.19.0","npm":"6.14.4"})
npm WARN notsup Not compatible with your version of node/npm: schema-utils@4.0.0
npm WARN exampleSite@0.1.0 No description
npm WARN exampleSite@0.1.0 No repository field.
npm WARN exampleSite@0.1.0 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

> added 481 packages from 265 contributors and audited 484 packages in 16.81s

Then when I tried the server ...

`hugo server`

> Start building sites … 
hugo v0.94.2-48FB9E4D linux/amd64 BuildDate=2022-03-12T10:28:42Z VendorInfo=gohugoio
Error: Error building site: TOCSS: failed to transform "katex/scss/index.scss" (text/x-scss). Check your Hugo installation; you need the extended version to build SCSS/SASS.: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
Built in 1575 ms

`hugo version`

> hugo v0.94.2-48FB9E4D linux/amd64 BuildDate=2022-03-12T10:28:42Z VendorInfo=gohugoio

**Reply by razonyang** [reply]
That's right, the **extended** version is required. You can download the extended version from [GitHub Releases](https://github.com/gohugoio/hugo/releases/tag/v0.94.0).

**Reply by razonyang** [reply]
BTW, the following configuration(`config.toml`) must be set:

```toml
[build]
  writeStats = true
```

It tells Hugo to generate the stats file used by `postcss` for purging unused CSS.

**Reply by jehoshua7** [reply]
@razonyang - Thanks, even with Hugo extended, it still throws an error

`cd myblog`
`hugo server`

> Start building sites … 
hugo v0.94.0-63B23660+extended linux/amd64 BuildDate=2022-03-10T09:46:36Z VendorInfo=gohugoio
Error: Error building site: POSTCSS: failed to transform "assets/main/bundle.min.css" (text/css): /home/********/Hugo/myblog/node_modules/yargs-parser/build/index.cjs:1013
        throw Error(`yargs parser supports a minimum Node.js version of ${minNodeVersion}. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions`);
        ^

> Error: yargs parser supports a minimum Node.js version of 12. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions
    at Object.<anonymous> (/home/********/Hugo/myblog/node_modules/yargs-parser/build/index.cjs:1013:15)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/********/Hugo/myblog/node_modules/yargs/build/index.cjs:1:58028)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
Built in 5850 ms

`hugo version`

> hugo v0.94.0-63B23660+extended linux/amd64 BuildDate=2022-03-10T09:46:36Z VendorInfo=gohugoio

**Reply by razonyang** [reply]
I guess your Node version is too low, please try to upgrade the Node and try again.
https://nodejs.org/en/download/

**Comment by jehoshua7** [comment]
For me, it will be easier to go back to https://github.com/razonyang/hugo-theme-bootstrap/releases/tag/v0.67.1

No NPM it seems.

**Reply by Natenom** [reply]
I had similiar issues on a Debian server with old NPM versions. I ended up with using NVM (Node Version Manager) and got the latest version. It works well now.
